toolchain/check/declaration_name_stack.cpp

```diff
@@ -8,11 +8,22 @@
 
 namespace Carbon::Check {
 
+auto DeclarationNameStack::NewNameContext() -> NameContext {
+  return NameContext{.state = NameContext::State::New,
+                     .target_scope_id = context_->current_scope_id(),
+                     .resolved_node_id = SemIR::NodeId::Invalid};
+}
```

Comment:
What if you were to add a constructor to NameContext that took the scope as an argument, and just pass that rather than this function?

Otherwise, "New" makes me think `new` allocation; I think Carbon's design is heading towards "Make" so that might be fair, or so would "Build". Alternatively, maybe you're trying to match the enum name: rename it and this to "Start"? (State::Start, StartNameContext)

Reply:
How about `State::Empty` and `MakeEmptyNameContext`?

Reply:
Sure